python/cudf/cudf/core/dataframe.py

```diff
@@ -6239,6 +6240,146 @@ def interleave_columns(self):
             {None: libcudf.reshape.interleave_columns([*self._columns])}
         )
 
+    @_cudf_nvtx_annotate
+    def eval(self, expr: "str", inplace: "bool" = False, **kwargs):
```

Comment:
how worthwhile is it to match the pandas signature here? It seems like it supports some bells and whistles that I am guessing we don't and won't in the near term at least.

Reply:
Unrelated to Brandon's comment above: we shouldn't need to use string type annotations for built-in types.
```suggestion
    def eval(self, expr: str, inplace: bool = False, **kwargs):
```

Reply:
> how worthwhile is it to match the pandas signature here? It seems like it supports some bells and whistles that I am guessing we don't and won't in the near term at least.

I'm assuming that this question is specifically about the `**kwargs` and not `inplace`, which is supported now. I originally had an error message if kwargs was non-empty, but I guess I accidentally removed that. I figured I would match the signature but throw an error if kwargs was non-empty, which is consistent with how we deal with unsupported pandas parameters n various other parts of the code as well. Does that sound reasonable @brandon-b-miller? Regarding specifics, I think there are some parameters to pd.eval that we could support, but certainly not all of them.

The annotations are the way that they are because they're copied from pandas, I can definitely change them now.